### ISSUE TITLE
🐛 [firebase_crashlytics] Android - Updating the app with firebase_crashlytics 0.2.0-dev5 causing android.database.sqlite.SQLiteException: table event_payloads already exists

### ISSUE THREAD

**Issue by red42** [FirebaseExtended/flutterfire#3530]
## Bug report

Some of my app's users started reporting the following error after updating to a version using firebase_crashlytics 0.2.0-dev.5:
```
Fatal Exception: android.database.sqlite.SQLiteException: table event_payloads already exists (code 1 SQLITE_ERROR[1]): , while compiling: CREATE TABLE event_payloads (sequence_num INTEGER NOT NULL, event_id INTEGER NOT NULL, bytes BLOB NOT NULL,FOREIGN KEY (event_id) REFERENCES events(_id) ON DELETE CASCADE,PRIMARY KEY (sequence_num, event_id))
```

### Steps to reproduce

Most (possibly all) cases happened after the app was updated. New installations seem fine, which makes sense given the type of error.

### Expected behavior

According to [this report on Firebase Android SDK](https://github.com/firebase/firebase-android-sdk/issues/1640), the bug was fixed in crashlytics 17.0.1. The current BoM for firebase_core is 25.3.1, which maps to crashlytics 17.0.0.

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.6 19G2021, locale en-BR)
    • Flutter version 1.20.2 at /Users/rafaelmatsunaga/flutter
    • Framework revision bbfbf1770c (4 weeks ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/rafaelmatsunaga/Library/Android/sdk
    • Platform android-29, build-tools 30.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.1

[!] Connected device
    ! No devices available
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.9.1
Flutter SDK 1.20.2
mangos 2020.09.10+20091001

dependencies:
- advertising_id 0.9.2 [flutter]
- cached_network_image 2.3.1 [flutter flutter_cache_manager octo_image]
- camera 0.5.7+5 [flutter]
- carousel_slider 2.2.1 [flutter]
- cloud_firestore 0.14.0+2 [flutter meta quiver firebase_core firebase_core_platform_interface cloud_firestore_platform_interface cloud_firestore_web]
- cloud_functions 0.6.0 [meta flutter firebase_core cloud_functions_platform_interface cloud_functions_web]
- data_connection_checker 0.3.4
- device_info 0.4.2+4 [flutter]
- dotted_border 1.0.6 [flutter path_drawing]
- equatable 1.2.5 [collection meta]
- fast_qr_reader_view 0.1.8 [flutter]
- firebase_analytics 6.0.0 [meta flutter firebase_core firebase_analytics_web firebase_analytics_platform_interface]
- firebase_auth 0.18.0+1 [meta firebase_core firebase_core_platform_interface firebase_auth_platform_interface firebase_auth_web flutter]
- firebase_auth_oauth 0.2.1 [flutter firebase_auth_oauth_platform_interface firebase_auth_oauth_web firebase_auth firebase_core]
- firebase_core 0.5.0 [firebase_core_platform_interface flutter quiver meta firebase_core_web]
- firebase_crashlytics 0.2.0-dev.5 [flutter stack_trace firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface]
- firebase_dynamic_links 0.6.0 [flutter firebase_core]
- firebase_in_app_messaging 0.2.0+1 [meta flutter firebase_core]
- firebase_messaging 7.0.0 [meta platform flutter firebase_core]
- firebase_performance 0.4.0 [flutter firebase_core]
- firebase_remote_config 0.4.0 [flutter firebase_core]
- firebase_storage 4.0.0 [flutter firebase_core]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_bloc 6.0.5 [flutter bloc provider]
- flutter_cache_manager 1.4.1 [flutter path_provider uuid http path sqflite pedantic clock file rxdart]
- flutter_facebook_login 3.0.0 [flutter collection]
- flutter_svg 0.18.1 [path_drawing xml vector_math meta flutter]
- flutter_typeahead 1.8.8 [flutter flutter_keyboard_visibility]
- flutter_widget_from_html_core 0.5.0+3 [flutter html]
- geocoding 1.0.3 [flutter geocoding_platform_interface url_launcher]
- geohash 0.2.1
- geolocator 6.0.0+4 [flutter geolocator_platform_interface]
- google_maps_flutter 0.5.31 [flutter flutter_plugin_android_lifecycle google_maps_flutter_platform_interface]
- google_sign_in 4.5.3 [google_sign_in_platform_interface flutter meta google_sign_in_web]
- http 0.12.2 [http_parser path pedantic]
- image 2.1.14 [archive xml]
- intl 0.16.1 [path]
- map_launcher 0.12.2 [flutter flutter_svg]
- package_info 0.4.1 [flutter]
- path_provider 1.6.14 [flutter path_provider_platform_interface path_provider_macos path_provider_linux]
- rxdart 0.24.1
- share 0.6.4+3 [meta flutter]
- sprintf 4.1.0
- unorm_dart 0.1.2
- url_launcher 5.5.2 [flutter url_launcher_platform_interface url_launcher_web url_launcher_linux url_launcher_macos]
- video_player 0.10.11+2 [meta video_player_platform_interface video_player_web flutter]

dev dependencies:
- flutter_launcher_icons 0.7.5 [image args yaml]
- flutter_launcher_name 0.0.1 [flutter yaml]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]
- test 1.15.2 [analyzer async boolean_selector coverage http http_multi_server io js node_preamble package_config path pedantic pool shelf shelf_packages_handler shelf_static shelf_web_socket source_span stack_trace stream_channel typed_data web_socket_channel webkit_inspection_protocol yaml test_api test_core]

transitive dependencies:
- _fe_analyzer_shared 6.0.0 [meta]
- analyzer 0.39.16 [_fe_analyzer_shared args charcode cli_util collection convert crypto glob html meta package_config path pub_semver source_span watcher yaml]
- archive 2.0.13 [crypto args path]
- args 1.6.0
- async 2.4.2 [collection]
- bloc 6.0.1 [meta]
- boolean_selector 2.0.0 [source_span string_scanner]
- characters 1.0.0
- charcode 1.1.3
- cli_util 0.1.4 [path]
- clock 1.0.1 [meta]
- cloud_firestore_platform_interface 2.0.1 [flutter meta collection firebase_core plugin_platform_interface]
- cloud_firestore_web 0.2.0+1 [flutter flutter_web_plugins firebase http_parser meta firebase_core cloud_firestore_platform_interface js]
- cloud_functions_platform_interface 2.0.0 [flutter meta firebase_core plugin_platform_interface]
- cloud_functions_web 2.0.0 [cloud_functions_platform_interface flutter flutter_web_plugins firebase http_parser meta]
- collection 1.14.13
- convert 2.1.1 [charcode typed_data]
- coverage 0.14.0 [args logging package_config path source_maps stack_trace vm_service]
- crypto 2.1.5 [collection convert typed_data]
- csslib 0.16.2 [source_span]
- fake_async 1.1.0 [clock collection]
- file 5.2.1 [intl meta path]
- firebase 7.3.0 [http http_parser js]
- firebase_analytics_platform_interface 1.0.3 [flutter meta]
- firebase_analytics_web 0.1.1 [flutter flutter_web_plugins firebase firebase_analytics_platform_interface meta]
- firebase_auth_oauth_platform_interface 0.2.1 [flutter firebase_auth firebase_core]
- firebase_auth_oauth_web 0.2.1 [flutter flutter_web_plugins firebase_auth_oauth_platform_interface firebase js firebase_auth firebase_core]
- firebase_auth_platform_interface 2.0.1 [flutter meta firebase_core plugin_platform_interface]
- firebase_auth_web 0.3.0+1 [flutter flutter_web_plugins firebase meta http_parser intl firebase_core firebase_auth_platform_interface js]
- firebase_core_platform_interface 2.0.0 [flutter meta plugin_platform_interface quiver]
- firebase_core_web 0.2.0 [firebase firebase_core_platform_interface flutter flutter_web_plugins meta js]
- firebase_crashlytics_platform_interface 1.0.0-dev.2 [flutter meta collection firebase_core plugin_platform_interface]
- flutter_blurhash 0.5.0 [flutter meta]
- flutter_keyboard_visibility 3.2.1 [flutter]
- flutter_plugin_android_lifecycle 1.0.8 [flutter]
- flutter_web_plugins 0.0.0 [flutter characters collection meta typed_data vector_math]
- geocoding_platform_interface 1.0.0+3 [flutter meta plugin_platform_interface]
- geolocator_platform_interface 1.0.4 [flutter meta plugin_platform_interface vector_math]
- glob 1.2.0 [async collection node_io path pedantic string_scanner]
- google_maps_flutter_platform_interface 1.0.4 [flutter meta plugin_platform_interface stream_transform]
- google_sign_in_platform_interface 1.1.2 [flutter meta quiver]
- google_sign_in_web 0.9.1+1 [google_sign_in_platform_interface flutter flutter_web_plugins meta js]
- html 0.14.0+3 [csslib source_span]
- http_multi_server 2.2.0 [async]
- http_parser 3.1.4 [charcode collection source_span string_scanner typed_data]
- io 0.3.4 [charcode meta path string_scanner]
- js 0.6.2
- logging 0.11.4
- matcher 0.12.8 [stack_trace]
- meta 1.1.8
- mime 0.9.7
- nested 0.0.4 [flutter]
- node_interop 1.1.1 [js]
- node_io 1.1.1 [node_interop path]
- node_preamble 1.4.12
- octo_image 0.2.1 [flutter flutter_blurhash]
- package_config 1.9.3 [path charcode]
- path 1.7.0
- path_drawing 0.4.1+1 [vector_math meta path_parsing flutter]
- path_parsing 0.1.4 [vector_math meta]
- path_provider_linux 0.0.1+2 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 0.0.4+3 [flutter]
- path_provider_platform_interface 1.0.2 [flutter meta platform plugin_platform_interface]
- pedantic 1.9.0
- petitparser 3.0.4 [meta]
- platform 2.2.1
- platform_detect 1.4.0 [meta pub_semver]
- plugin_platform_interface 1.0.2 [meta]
- pool 1.4.0 [async stack_trace]
- process 3.0.13 [file intl meta path platform]
- provider 4.3.2 [flutter nested collection]
- pub_semver 1.4.4 [collection]
- quiver 2.1.3 [matcher meta]
- shelf 0.7.7 [async collection http_parser path stack_trace stream_channel]
- shelf_packages_handler 2.0.0 [path shelf shelf_static]
- shelf_static 0.2.8 [convert http_parser mime path shelf]
- shelf_web_socket 0.2.3 [shelf web_socket_channel stream_channel]
- sky_engine 0.0.99
- source_map_stack_trace 2.0.0 [path stack_trace source_maps]
- source_maps 0.10.9 [source_span]
- source_span 1.7.0 [charcode collection meta path term_glyph]
- sqflite 1.3.1 [flutter sqflite_common path]
- sqflite_common 1.0.2+1 [synchronized path meta]
- stack_trace 1.9.5 [path]
- stream_channel 2.0.0 [async]
- stream_transform 1.2.0
- string_scanner 1.0.5 [charcode meta source_span]
- synchronized 2.2.0+2
- term_glyph 1.1.0
- test_api 0.2.17 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- test_core 0.3.10 [analyzer async args boolean_selector collection coverage glob io meta package_config path pedantic pool source_map_stack_trace source_maps source_span stack_trace stream_channel vm_service yaml matcher test_api]
- typed_data 1.2.0 [collection]
- url_launcher_linux 0.0.1+1 [flutter]
- url_launcher_macos 0.0.1+7 [flutter]
- url_launcher_platform_interface 1.0.8 [flutter meta plugin_platform_interface]
- url_launcher_web 0.1.3+1 [url_launcher_platform_interface platform_detect flutter flutter_web_plugins meta]
- uuid 2.2.0 [crypto convert]
- vector_math 2.0.8
- video_player_platform_interface 2.0.2 [flutter meta]
- video_player_web 0.1.3+2 [flutter flutter_web_plugins meta video_player_platform_interface]
- vm_service 4.2.0 [meta]
- watcher 0.9.7+15 [async path pedantic]
- web_socket_channel 1.1.0 [async crypto stream_channel]
- webkit_inspection_protocol 0.7.3 [logging]
- xdg_directories 0.1.0 [path process flutter]
- xml 4.2.0 [collection convert meta petitparser]
- yaml 2.2.1 [charcode collection string_scanner source_span]
```

</details>

---

**Comment by red42** [comment]
As a workaround, overriding FirebaseSDKVersion to version 25.4.1 (first version with crashlytics 17.0.1) seems to fix the problem.

Just add the following to android/app/build.gradle:
```
project.ext {
    set('FlutterFire', [
            FirebaseSDKVersion: '25.4.1'
    ])
}
```

**Comment by serhii-k** [comment]
Should be closed for now, as firebase_core 0.5.1 now sets FirebaseSDKVersion=25.12.0.
Thank you very much @red42 for your valuable suggestions!

**Comment by red42** [comment]
It seems to be fixed indeed! Thank you!
